identity_iota_core/src/state_metadata/document.rs

```diff
@@ -93,26 +95,47 @@ impl StateMetadataDocument {
       identity_did::Error::InvalidDocument("expected data to have at least length 5", None),
     )?)?;
 
-    let inner: &[u8] = data.get(5..).ok_or(identity_did::Error::InvalidDocument(
-      "expected data to have at least length 6",
-      None,
-    ))?;
+    let data_len_packed: [u8; 2] = data
+      .get(5..=6)
```

Comment:
This style seems better. feel free to change the `.get(0..3)` line for the marker as well.